File: examples/wasm-emscripten/sample.js
function fibonacci (n) {
    if (n <= 0) {
        return 0;
    } else if (n == 1) {
        return 1;
    }
    return fibonacci(n - 1) + fibonacci(n - 2);
}

function pi (nPoints) {
   var a, b, inside = 0;
    for (var i = 0; i < nPoints; i++) {
        a = Math.random();
        b = Math.random();
        if (a*a + b*b <= 1.0) {
           ++inside;
        }
    }
    return inside / nPoints * 4.0;
}

exports.fibonacci = fibonacci;
exports.pi = pi;

File: examples/wasm-emscripten/sample.c
#include <stdlib.h>
#include <time.h>
#include <math.h>
#include <emscripten/emscripten.h>

EMSCRIPTEN_KEEPALIVE
int fibonacci (int n) {
    if (n <= 0) {
        return 0;
    } else if (n == 1) {
        return 1;
    }
    return fibonacci(n - 1) + fibonacci(n - 2);
}

EMSCRIPTEN_KEEPALIVE
double pi (int nPoints) {
    int i, inside = 0;
    double a, b;

    srand(time(NULL));
    for (i = 0; i < nPoints; i++) {
        a = (double)rand() / RAND_MAX;
        b = (double)rand() / RAND_MAX;
        if (a*a + b*b <= 1.0) {
            ++inside;
        }
    }
    return (double)inside / nPoints * 4.0;
}
